core/base/dispatch_helper.hpp

```diff
@@ -193,10 +236,9 @@ auto run(std::shared_ptr<T> obj, Func&& f, Args&&... args)
  * template and run it.
  *
  * @tparam Base  the Base class with one template
- * @tparam K  the current template type of B. pointer of const Base<K> is tried
- *            in the conversion.
- * @tparam ...Types  the other types will be tried in the conversion if K fails
- * @tparam T  the type of input object waiting converted
+ * @tparam ...Types  the types that will be tried with Base, i.e. Base<Types>...
+ * @tparam T  the type of input object waiting converted, has to be either
+ *            unique_ptr or shared_ptr
```

Comment:
It can only be `shared_ptr` because `dynamic_pointer_cast` only allow `shared_ptr` and the ownership issue.